kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +313,77 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps) throws KuraException {
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            if (prop.getKey().contains("802-1x.keystore.pid")) {
+                String keystorePid = (String) prop.getValue();
+
+                String interfaceName = prop.getKey().split("\\.")[2];
+                findAndDecodeCertificatesForInterface(interfaceName, modifiedProps, getKeystore(keystorePid));
+            }
+        }
+    }
+
+    private void findAndDecodeCertificatesForInterface(String interfaceName, Map<String, Object> modifiedProps,
+            KeystoreService keystoreService) {
+
+        if (keystoreService == null) {
+            logger.error("Cannot find keystore service for interface {}", interfaceName);
+            return;
+        }
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            String clientCertString = String.format("%s.config.802-1x.client-cert-name", interfaceName);
+            String caCertString = String.format("%s.config.802-1x.802-1x.ca-cert-name", interfaceName);
+            String privateKeyString = String.format("%s.config.802-1x.private-key-name", interfaceName);
+            if (prop.getKey().contains(clientCertString) || prop.getKey().contains(caCertString)) {
+                Object value = prop.getValue();
+                try {
+                    String valueString = (String) prop.getValue();
+                    modifiedProps.put(prop.getKey(), decryptCertificate(valueString, keystoreService));
+                } catch (Exception e) {
+                    logger.error("Enable to decode certificate {} from keystore.", value.toString(), e);
+                    modifiedProps.put(prop.getKey(), value);
+                }
+            } else if (prop.getKey().contains(privateKeyString)) {
+                Object value = prop.getValue();
+                try {
+                    String valueString = (String) prop.getValue();
+                    modifiedProps.put(prop.getKey(), decryptPrivateKey(valueString, keystoreService));
+                } catch (Exception e) {
+                    logger.error("Enable to decode private key {} from keystore.", value.toString(), e);
```

Comment:
No need to call "toString()" method as formatting and string conversion is done by the Formatter

```suggestion
                    logger.error("Enable to decode private key {} from keystore.", value, e);
```